### PR TITLE
Take 2 of multiple environments for dotenv-rails

### PR DESCRIPTION
**Pull request by bkeepers**
This replaces https://github.com/bkeepers/dotenv/pull/120 and closes https://github.com/bkeepers/dotenv/issues/111 and closes https://github.com/bkeepers/dotenv/issues/109 and closes https://github.com/bkeepers/dotenv/issues/45.

This is a pretty simple approach that only loads `.env.local`, `.env.#{Rails.env}`, and `.env` (first one to set a variable wins) that does not override any existing environment variables.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,6 +101,14 @@ Whenever your application loads, these variables will be available in `ENV`:
 config.fog_directory  = ENV['S3_BUCKET']
 ```
 
+## Multiple Rails Environments
+
+dotenv was originally created to load configuration variables into `ENV` in *development*. There are typically better ways to manage configuration in production environments environments—such as `/etc/environment` managed by [Puppet](https://github.com/puppetlabs/puppet) or [Chef](https://github.com/opscode/chef), `heroku config`, etc.
+
+However, some find dotenv to be a convenient way to configure Rails applications in staging and production environments, and you can do that by defining environment-specific files like `.env.production` or `.env.test`.
+
+You can also `.env.local` for local overrides.
```
Comment (adrianmacneil) [inline README.md]: typo

You can also *use* `.env.local`

**Comment by adrianmacneil** [comment]
:+1: looks great

**Comment by jjb** [comment]
:dancers:

**Comment by ericboehs** [comment]
Just thought I'd point out that [invoker](https://github.com/code-mancers/invoker/search?utf8=%E2%9C%93&q=.env) also does `.env` importing. If you're using dotenv in conjunction, the load order gets messed up and `.env` take precedence over `.env.local` (the opposite of this PR). It's an issue with invoker and not dotenv.

**Pull request merged**